syntax = "proto3";

package gredis;

service Gredis {
  // Get the value stored in redis for a given key
  rpc GetValue (GetValueRequest) returns (GetValueReply) {}

  // Set the redis's value for a given key
  rpc SetValue (SetValueRequest) returns (SetValueReply) {}
}

message GetValueRequest {
  string key = 1;
}

message GetValueReply {
  string value = 1;
}

message SetValueRequest {
  string key = 1;
  string value = 2;
}

message SetValueReply {
  // Returns the value.
  string value = 1;
}
